### PR TITLE
Allow paging of the index query

### PR DESCRIPTION
**Pull request by jamsilver**

### CODE REVIEW

File: includes/profile.inc [hunk 1]
```diff
@@ -580,12 +587,16 @@ function cm_tools_tables_walk(array &$sandbox, $tables, callable $query_alter_ca
       if (is_callable($query_alter_callback)) {
         call_user_func($query_alter_callback, $query, $table_name);
       }
+      if (isset($index_query_page_size)) {
+        $query->range($s['page'] * $index_query_page_size, $index_query_page_size);
```
Comment (jamsilver) [inline includes/profile.inc]: @darthsteven I assume we _can_ rely on mysql to return rows in a consistent order such that successive invocations of this query perfectly segment the whole table?
* `... LIMIT 0, 4000`
* `... LIMIT 4000, 4000`
* `... LIMIT 8000, 4000`
* `... LIMIT 12000, 4000`
??

Comment (jamsilver) [reply includes/profile.inc]: Actually, if I add an ORDER BY that includes all the primary columns, then the order should be defined anyway, so: yes this should work.🤔

Comment (jamsilver) [reply includes/profile.inc]: Done

Comment (darthsteven) [reply includes/profile.inc]: > @darthsteven I assume we _can_ rely on mysql to return rows in a consistent order such that successive invocations of this query perfectly segment the whole table?
> 
> * `... LIMIT 0, 4000`
> * `... LIMIT 4000, 4000`
> * `... LIMIT 8000, 4000`
> * `... LIMIT 12000, 4000`
>   ??

Unless you have an ORDER BY, no, you cannot make that assumption :)

**Review by darthsteven — APPROVED** [review]

**Pull request merged**